blackjax/mcmc/elliptical_slice.py

```diff
@@ -0,0 +1,166 @@
+    theta
+        A value between [-2\pi, 2\pi] identifying points in the ellipsis drawn
+        from the positon and mommentum variables. This value indicates the theta
+        value of the accepted proposal.
+    subiter
+        Number of sub iterations needed to accept a proposal. The more subiterations
+        needed the less efficient the algorithm will be, and the more dependent the
+        new value is likely to be to the previous value.
+    """
+
+    momentum: PyTree
+    theta: float
+    subiter: int
+
+
+def init(position: PyTree, loglikelihood_fn: Callable):
+    loglikelihood = loglikelihood_fn(position)
+    return EllipSliceState(position, loglikelihood)
+
+
+def kernel(cov_matrix: Array, mean: Array):
+    """Build an Elliptical Slice sampling kernel.
+
+    Parameters
+    ----------
+    cov_matrix
+        The value of the covariance matrix of the gaussian prior distribution from the posterior we wish to sample.
+
+    Returns
+    -------
+    A kernel that takes a rng_key and a Pytree that contains the current state
+    of the chain and that returns a new state of the chain along with
+    information about the transition.
+
+    """
+
+    cov_matrix_sqrt = jax.lax.linalg.cholesky(cov_matrix)
+
+    def momentum_generator(rng_key, position):
+        p, unravel_fn = jax.flatten_util.ravel_pytree(position)
+        momentum = mean + jnp.dot(
+            cov_matrix_sqrt, jax.random.normal(rng_key, shape=p.shape)
+        )
+        return unravel_fn(momentum)
+
+    def one_step(
+        rng_key: PRNGKey,
+        state: EllipSliceState,
+        loglikelihood_fn: Callable,
+    ) -> Tuple[EllipSliceState, EllipSliceInfo]:
+        proposal_generator = elliptical_proposal(
+            loglikelihood_fn, momentum_generator, mean
+        )
+        return proposal_generator(rng_key, state)
+
+    return one_step
+
+
+def elliptical_proposal(
+    loglikelihood_fn: Callable,
+    momentum_generator: Callable,
+    mean: Array,
+) -> Callable:
+    """Build an Ellitpical slice sampling kernel.
+
+    The algorithm samples a latent parameter, traces an ellipse connecting the
+    initial position and the latent parameter and does slice sampling on this
+    ellipse to output a new sample from the posterior distribution.
+
+    Parameters
+    ----------
+    loglikelihood_fn
+        A function that returns the log-likelihood at a given position.
+    momentum_generator
+        A function that generates a new latent momentum variable.
+
+    Returns
+    -------
+    A kernel that takes a rng_key and a Pytree that contains the current state
+    of the chain and that returns a new state of the chain along with
+    information about the transition.
+
+    """
+
+    def generate(
+        rng_key: PRNGKey, state: EllipSliceState
+    ) -> Tuple[EllipSliceState, EllipSliceInfo]:
+        position, loglikelihood = state
+        kmomentum, kunif, ktheta = jax.random.split(rng_key, 3)
```

Comment:
We tend to prefer explicit names to shortcuts when it doesn't make the code harder to read. 

`kery_momentum`, `key_uniform` and `key_theta`?